File: functions/libsodium/variants/go/2_b64_char_to_byte_9.go
package main

func b64CharToByte(c int32) uint32 {
	if c >= 65 && c <= 90 {
		return uint32(c - 65)
	}
	if c >= 97 && c <= 122 {
		return uint32(c - 97 + 26)
	}
	if c >= 48 && c <= 57 {
		return uint32(c - 48 + 52)
	}
	if c == 43 {
		return 62
	}
	if c == 47 {
		return 63
	}
	return 0
}


func main(){
    b64CharToByte(0)
}

File: functions/libsodium/variants/c/2_b64_char_to_byte_9.c
#include <stdint.h>

static unsigned int b64_char_to_byte(int c)
{
    return c >= 'A' && c <= 'Z' ? c - 'A' :
           c >= 'a' && c <= 'z' ? c - 'a' + 26 :
           c >= '0' && c <= '9' ? c - '0' + 52 :
           c == '+' ? 62 :
           c == '/' ? 63 : 0;
}


int main(){
    b64_char_to_byte(0);
}
